packages/app-store/office365calendar/lib/CalendarService.ts

```diff
@@ -689,17 +690,49 @@ export default class Office365CalendarService implements Calendar {
   async getMainTimeZone(): Promise<string> {
     try {
       const response = await this.fetcher(`${await this.getUserEndpoint()}/mailboxSettings/timeZone`);
-      const timezone = await handleErrorsJson<string>(response);
+      const timezoneResponse = await handleErrorsJson<string | { value: string }>(response);
 
-      if (!timezone || typeof timezone !== "string") {
+      let windowsTimezoneName: string;
+
+      if (typeof timezoneResponse === "object" && timezoneResponse !== null && "value" in timezoneResponse) {
+        windowsTimezoneName = timezoneResponse.value;
```

Comment:
Also, the docs linked with possible response types
1. https://learn.microsoft.com/en-us/graph/api/user-get-mailboxsettings?view=graph-rest-1.0&tabs=http#response-4
2. https://learn.microsoft.com/en-us/graph/api/user-get-mailboxsettings?view=graph-rest-1.0&tabs=javascript#response-1

have timeZone key in the response, but the code seemingly expects object with `{value: string}` type coming from api.